src/index.ts

```diff
@@ -50,13 +87,14 @@ app.post("/", verifySignatureMiddleware, express.json(), async (req, res) => {
         "<-- END OF LIST OF MODELS -->",
       ].join("\n"),
     },
-    ...req.body.messages,
-  ].concat(req.body.messages);
+    ...payload.messages,
+  ].concat(payload.messages);
 
   console.time("tool-call");
   const toolCaller = await capiClient.chat.completions.create({
     stream: false,
     model: "gpt-4",
+    // @ts-expect-error - TODO @gr2m - type incompatibility between @openai/api and @copilot-extensions/preview-sdk
     messages: toolCallMessages,
```

Comment:
Why? Why wouldn't we make the preview-sdk's types compatible?

Moreover: I'm not sure I agree that the SDK should contain an SDK for OpenAI. Like in #6, we're also calling the GitHub Models API - that's an OpenAI-compatible API, and has nothing to do with Copilot. I wouldn't expect or want to use a Copilot SDK for that.

Reply:
the `prompt()` method originated from discussions in GitHub Models, where we wanted to simplify the getting started documentation by using our own SDK that removes some of the complexities. However that related to Azure's SDKs.

I can bring back the OpenAI SDK if you prefer, it's just quite a heavy dependency with a lot of functionality you don't need here

Reply:
I removed the `@ts-expect-error` in 7507a09. If anything it belongs into #6.

Reply:
> it's just quite a heavy dependency with a lot of functionality you don't need here

I don't really agree - and it's more a question of what makes most sense for _the models extension_ vs. all Copilot Extensions. As an Extension that calls multiple models, the more robust and well-used SDK feels more appropriate. Besides which, for an example repository I'd bias more towards clarity and adherence to the ecosystem over performance or dependency weight.